precompiles/utils/src/data/mod.rs

```diff
@@ -431,6 +431,49 @@ impl EvmData for U256 {
 	}
 }
 
+impl<T> EvmData for Option<T>
```

Comment:
Please remove this bad implementation. You don't need it.

I discussed this with @Jérémy, and the format is not compatible with substrate behavior.

There are no optional parameters in solidity, either you have to use a special value (0x000 or 0xfff) or you have to create a custom format with solidity type `bytes` to be sure to parse it correctly.